### PR TITLE
add patch CLI

### PR DESCRIPTION
**Pull request by lukasheinrich**
# Description

This adds a `--patch` flag to `pyhf cls`

```
cat newsignal.json|pyhf cls original.json --patch -
```

```
pyhf cls original.json --patch newsignal.json
```

```
pyhf cls original.json --patch newsignal.json
```

```
cat original.json pyhf cls - --patch newsignal.json
```

# Checklist Before Requesting Approver

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request

### CODE REVIEW

**Comment by kratsg** [comment]
Saw this in your slides! Neat :)

File: pyhf/commandline.py [hunk 1]
```diff
@@ -5,12 +5,14 @@
 import click
 import json
 import os
+import jsonpatch
```
Comment (kratsg) [inline pyhf/commandline.py]: We need the additional setup.py requirements for this dependency, no?

Comment (lukasheinrich) [reply pyhf/commandline.py]: added. schould be fine. only sub-dependency is jsonpinter which is in turn python-stdlib only

**Comment by lukasheinrich** [comment]
need to add tests

**Comment by matthewfeickert** [comment]
I think once tests are in this will be an awesome addition! :)

File: tests/test_scripts.py [hunk 2]
```diff
@@ -68,3 +68,38 @@ def test_import_and_export(tmpdir, script_runner):
 
     command = 'pyhf json2xml {0:s} --specroot {1:s} --dataroot {1:s}'.format(temp.strpath,str(tmpdir))
     ret = script_runner.run(*shlex.split(command))
+    assert ret.success
```
Comment (lukasheinrich) [inline tests/test_scripts.py]: drive-by fix

**Comment by matthewfeickert** [comment]
Seems there is some pesky Python 2.7 failure still though :disappointed:  (he said annoyingly stating the obvious). Python 3 is perfectly happy!

File: tests/test_scripts.py (hunk 2 above)
Comment (matthewfeickert) [reply tests/test_scripts.py]: [pytest is still giving some errors in Python 2.7](https://travis-ci.org/diana-hep/pyhf/jobs/428334799#L510-L512)

```
>       assert ret.success
E       assert False
E        +  where False = <pytest_console_scripts.RunResult object at 0x7f80a9d50e50>.success
```

Comment (lukasheinrich) [reply tests/test_scripts.py]: something weird in the way the `pytest` file tmp file objects work in 2.7. They should be file-like but they are not. Changed it to `io.StringIO`

Comment (lukasheinrich) [reply tests/test_scripts.py]: see line 91
